### PR TITLE
Initial struct of python integration test

### PR DESCRIPTION
**Pull request by psy2848048**
## Updates
* Applied `pytest`
* Wrapped CLIs by python subprocess. It covers:
    * Simple one time execution
    * Multiple CLI interaction
    * Daemon

## Good to review
* Execute `pip3 install -r integration_tests/requirements`, then execute `cd integration_tests && pytest`
* @slave5vw This boilerplate will be applied to tendermint too. Your confirmation is necessary.
* Skipped `bonding/unbonding` and `create-validator`. Will be added proper testing criteria more than sending tx and checking successful.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -40,6 +42,12 @@ before_install:
   - bash scripts/travis_ci_prerequisites.sh
   - export PATH="$PATH:$HOME/.cargo/bin"
   - export PATH="$PATH:$HOME/protoc/bin"
+  - git clone https://github.com/pyenv/pyenv-virtualenv.git $(pyenv root)/plugins/pyenv-virtualenv
+  - echo 'eval "$(pyenv virtualenv-init -)"' >> ~/.bash_profile
+  - source ~/.bash_profile
+  - pyenv virtualenv 3.7.1 tests
+  - pyenv activate tests
+  - pip3 install -r integration_tests/requirements.txt
```
Comment (jhyeom26) [inline .travis.yml]: I am just curious that how about installing python3, python3-venv, python3-pip through addons?

```
addons:
  apt:
    packages:
      - curl
      - python3
      - python3-venv
      - python3-pip
  homebrew:
    package:
      - curl
      - python3 ......
```

Is it more verbose?

Comment (psy2848048) [reply .travis.yml]: `pyenv` and its plugin is more developed tool, and it's already installed in default Travis CI imgae.
Better to use it.

Comment (psy2848048) [reply .travis.yml]: Need to move these lines to Makefile?

File: .travis.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ jobs:
           - CasperLabs
       script:
         - make test
+        - cd integration_tests && pytest -s
```
[the remark this answers is not in the document]
Comment (psy2848048) [reply .travis.yml]: Thx for good idea :) Will do that

File: .travis.yml (hunk 1 above)
Comment (jhyeom26) [reply .travis.yml]: If this is travis specific thing, I think do not have to put them to Makefile, But if it is, we can move them to travis_ci_prerequisites.sh ;)

Comment (psy2848048) [reply .travis.yml]: As these packages and plugins are needed only in CI, will pour them into `travis_ci_prerequisites.sh`

File: .travis.yml [hunk 3]
```diff
@@ -30,12 +45,8 @@ jobs:
       script:
         - unset -f cd # workaround for https://github.com/travis-ci/travis-ci/issues/8703
         - make test
+        - make integration-tests
 
-addons:
-  apt:
-    packages: curl
-  homebrew:
-    package: curl
```
Comment (jhyeom26) [inline .travis.yml]: How about not moving addons's place~?

```
addons:
  apt:
    packages:
      - curl
      - python3-pip
  homebrew:
    package: curl
```

File: .travis.yml [hunk 4]
```diff
@@ -13,14 +21,21 @@ jobs:
...
       after_success:
         - bash <(curl -s https://codecov.io/bash)
 
     - stage: unit test
       os: osx
       osx_image: xcode11.3
-      language: go
+      language: 
+        - go
+        - shell
```
Comment (jhyeom26) [inline .travis.yml]: I am curious that why do we need the shell.

Comment (psy2848048) [reply .travis.yml]: Did as described at https://docs.travis-ci.com/user/languages/python/#running-python-tests-on-multiple-operating-systems
But let me try to get rid of that

Comment (jhyeom26) [reply .travis.yml]: Oh, I got it, then how about leaving a related issue link as a comment?

Comment (psy2848048) [reply .travis.yml]: Works well. Resolved

Comment (psy2848048) [reply .travis.yml]: Changed as what you said against the doc, but it works well :)

**Comment by psy2848048** [comment]
As discussion with Casperlabs, purse for staking and transfering are separated.
So the test need updating, and a new commit will be pushed accordingly.

**Comment by JoowonYun** [comment]
> As discussion with Casperlabs, purse for staking and transfering are separated.
> So the test need updating, and a new commit will be pushed accordingly.

How about merging this one and using another pull req? 
bond things already skip.

**Comment by psy2848048** [comment]
@JoowonYun Sure then :)

**Pull request merged**